src/plugins/chart_expressions/expression_gauge/public/expression_renderers/gauge_renderer.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { i18n } from '@kbn/i18n';
+import React, { memo } from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import { ThemeServiceStart } from '../../../../../core/public';
+import { KibanaThemeProvider } from '../../../../kibana_react/public';
+import { ExpressionRenderDefinition } from '../../../../expressions/common/expression_renderers';
+import { EXPRESSION_GAUGE_NAME, GaugeExpressionProps } from '../../common';
+import { getFormatService, getPaletteService, getThemeService } from '../services';
+
+import GaugeComponent from '../components/gauge_component';
+import './index.scss';
+const MemoizedChart = memo(GaugeComponent);
```

Comment:
This seems like a good place to lazy-load the actual renderer. Tyler is right - this is making a lot of code being loaded synchronously which was loaded async before

Reply:
Good point. Do you think we should do the same with heatmap plugin? (I think so)